### PR TITLE
Table serialization

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- All tables used for post-processing were never serialized and so when we were restarting, they were not restarted correctly

# Description of the solution

- Add a small helper function to use the boost serialization feature
- Serialize everythign

# How Has This Been Tested?

- Added a serialization test

# Future changes

- We should add the same serialization feature to ALL tables used everywhere. This can be done in one PR for physics.

### CODE REVIEW

File: include/core/utilities.h [hunk 1]
```diff
@@ -247,5 +250,37 @@ announce_string(const ConditionalOStream &pcout,
 }
 
 
+/**
+ * @brief Serializes a table using boost serialization feature
+ * the filename should contain the desired extension
+ *
+ * @param table The table to be serialized
+ * @param filename The file name  (including the extension) to be used
```
Comment (AmishgaAlphonius) [inline include/core/utilities.h:258]:
```suggestion
 * @param filename The file name (including the extension) to be used
```

File: source/solvers/navier_stokes_base.cc [hunk 2]
```diff
@@ -1610,6 +1610,43 @@ NavierStokesBase<dim, VectorType, DofsType>::read_checkpoint()
 
 
   multiphysics->read_checkpoint();
+
+  // Deserialize all of the post-processing tables that are currently used
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_base.cc:1614]:
```suggestion
  // Deserialize all post-processing tables that are currently used
```

File: source/solvers/navier_stokes_base.cc [hunk 3]
```diff
@@ -2119,6 +2156,44 @@ NavierStokesBase<dim, VectorType, DofsType>::write_checkpoint()
       std::string triangulationName = prefix + ".triangulation";
       tria->save(prefix + ".triangulation");
     }
+
+  // Serialize all of the post-processing tables that are currently used
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_base.cc:2160]:
```suggestion
  // Serialize all post-processing tables that are currently used
```

File: source/solvers/navier_stokes_base.cc [hunk 4]
```diff
@@ -2119,6 +2156,44 @@ NavierStokesBase<dim, VectorType, DofsType>::write_checkpoint()
...
+  {
+    const Parameters::PostProcessing post_processing =
+      simulation_parameters.post_processing;
+
```
Comment (AmishgaAlphonius) [inline source/solvers/navier_stokes_base.cc:2164]:
```suggestion
```

**Review by AmishgaAlphonius — APPROVED** [review]
Very neat! :)
I have very minor suggestions :)

**Review by lpsaavedra — APPROVED** [review]

**Pull request merged**